glommio/src/io/directory.rs

```diff
@@ -137,11 +137,10 @@ impl Directory {
     }
 }
 
-fn contains_dir(path: &Path) -> bool {
-    let mut iter = path.components().peekable();
-    let is_file = match iter.peek() {
-        Some(std::path::Component::Normal(_)) => true,
-        _ => false,
-    };
-    iter.count() > 1 || !is_file
+fn contains_dir<P: AsRef<Path>>(path: P) -> bool {
```

Comment:
I don't think that this internal API should be generic, as it has no real benefit, but the disadvantage of possible making compile times longer.